src/reconcile/rec_write.c

```diff
@@ -407,6 +407,17 @@ __wt_reconcile(WT_SESSION_IMPL *session, WT_REF *ref,
 	 */
 	WT_PAGE_LOCK(session, page);
 
+	/*
+	 * The state of the page can change after the check is made and between
+	 * the lock, check again to make sure the state is not changed in
```

Comment:
Can improve upon the comment, see if you like something like this:

"Though the caller checks for the page to be evict-able, there is a window between that check and the page lock where the page can become ineligible for eviction. We continue with the eviction of this page only if the page remains evict-able after taking the page lock."